### PR TITLE
Fixed #34173 --  Skipped saving sessions on 5xx responses.

### PR DESCRIPTION
**Pull request by SirAbhi13**
Fixed [#34173](https://code.djangoproject.com/ticket/34173).

Sessions won't be saved when status code is greater than and equal to 500.

### CODE REVIEW

File: tests/sessions_tests/tests.py [hunk 1]
```diff
@@ -727,6 +727,32 @@ def response_500(requset):
...
+            response = HttpResponse("Page not Found")
+            response.status_code = 404
+            request.session["hello"] = "world"
+            return response
+
+        request = self.request_factory.get("/")
+        SessionMiddleware(response_404)(request)
+
+        # The value was saved above.
+        self.assertIn("hello", request.session.load())
```
Comment (felixxm) [inline tests/sessions_tests/tests.py:754]: This pass without the patch. Additional tests are welcome but should be moved to a separate commit.

File: django/contrib/sessions/middleware.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ def process_response(self, request, response):
                     expires = http_date(expires_time)
                 # Save the session data and refresh the client cookie.
                 # Skip session save for 500 responses, refs #3881.
-                if response.status_code != 500:
+                if response.status_code < 500:
```
Comment (felixxm) [inline django/contrib/sessions/middleware.py:57]: The comment above needs to be updated.

**Review by felixxm — COMMENTED** [review]
@SirAbhi13 Thanks :+1:

File: tests/sessions_tests/tests.py (hunk 1 above)
Comment (SirAbhi13) [reply tests/sessions_tests/tests.py:754]: Got it. Will remove this test testing status code below 500.

**Comment by SirAbhi13** [comment]
Thank you for the review and the edits! 🚀

**Comment by felixxm** [comment]
Merged in 17472c3304050fbbf7a65976ab7b392184d081d1.
